src/doc/style-guide/src/items.md

```diff
@@ -421,9 +421,20 @@ Format associated types like type aliases. Where an associated type has a
 bound, put a space after the colon but not before:
 
 ```rust
-pub type Foo: Bar;
+type Foo: Bar;
 ```
 
+If an associated type declaration has no bounds, and has a `where` clause with
+a single entry, format the entire type declaration including the `where` clause
+on the same line if it fits:
+
+```rust
+type Item<'a> where Self: 'a;
+```
+
+If the `where` clause contains multiple entries, or the associated type has any
```

Comment:
Please make it clear that this is just for definitions (i.e. that there is no `=` here)

Also please double check this is consistent for function declarations.

Reply:
Based on conversation in the style team meeting, I'm going to:
- Extend this to also apply to function declarations
- Also allow associated type bounds
- *Explicitly* not allow `=` or function bodies (rather than implying it with the word "declaration")